channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -888,27 +907,26 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
              */
             function _beaconSender(payload: IPayloadData, oncomplete: OnCompleteCallback, sync?: boolean) {
                 let internalPayload = payload as IInternalPayloadData;
-                let data = internalPayload  && internalPayload.data;
-                if (isString(data) && data.length > 0) {
+                let data = internalPayload  && internalPayload.oriPayload;
+                if (isArray(data) && data.length > 0) {
                     // The sendBeacon method returns true if the user agent is able to successfully queue the data for transfer. Otherwise it returns false.
                     if (!_doBeaconSend(data, oncomplete)) {
```

Comment:
I think the `data` here still needs to be the `internalPayload.data` and not the array.